tests/filesystem.go

```diff
@@ -36,44 +37,32 @@ func mustRun(t *testing.T, command string, args ...string) []byte {
 }
 
 func prepareRootPartitionForPasswd(t *testing.T, partitions []*types.Partition) {
-	for _, p := range partitions {
-		if p.Label == "ROOT" {
-			dirs := []string{
-				filepath.Join(p.MountPath, "home"),
-				filepath.Join(p.MountPath, "usr", "bin"),
-				filepath.Join(p.MountPath, "usr", "sbin"),
-				filepath.Join(p.MountPath, "usr", "lib64"),
-				filepath.Join(p.MountPath, "etc"),
-			}
-			for _, dir := range dirs {
-				_ = os.MkdirAll(dir, 0755)
-			}
-
-			symlinks := []string{"lib64", "bin", "sbin"}
-			for _, symlink := range symlinks {
-				_ = os.Symlink(
-					filepath.Join(p.MountPath, "usr", symlink),
-					filepath.Join(p.MountPath, symlink))
-			}
-
-			copies := [][2]string{
-				{"/etc/ld.so.cache", filepath.Join(p.MountPath, "etc")},
-				{"/lib64/libblkid.so.1", filepath.Join(p.MountPath, "usr", "lib64")},
-				{"/lib64/libpthread.so.0", filepath.Join(p.MountPath, "usr", "lib64")},
-				{"/lib64/libc.so.6", filepath.Join(p.MountPath, "usr", "lib64")},
-				{"/lib64/libuuid.so.1", filepath.Join(p.MountPath, "usr", "lib64")},
-				{"/lib64/ld-linux-x86-64.so.2", filepath.Join(p.MountPath, "usr", "lib64")},
-				{"/lib64/libnss_files.so.2", filepath.Join(p.MountPath, "usr", "lib64")},
-				{"bin/amd64/id", filepath.Join(p.MountPath, "usr", "bin")},
-				{"bin/amd64/useradd", filepath.Join(p.MountPath, "usr", "sbin")},
-				{"bin/amd64/usermod", filepath.Join(p.MountPath, "usr", "sbin")},
-			}
+	mountPath := getRootLocation(partitions)
+	if mountPath == "" {
+		return
+	}
+	dirs := []string{
+		filepath.Join(mountPath, "home"),
+		filepath.Join(mountPath, "usr", "bin"),
+		filepath.Join(mountPath, "usr", "sbin"),
+		filepath.Join(mountPath, "usr", "lib64"),
+		filepath.Join(mountPath, "etc"),
+	}
+	for _, dir := range dirs {
+		_ = os.MkdirAll(dir, 0755)
+	}
 
-			for _, cp := range copies {
-				_, _ = exec.Command("cp", cp[0], cp[1]).CombinedOutput()
-			}
-		}
+	symlinks := []string{"lib64", "bin", "sbin"}
+	for _, symlink := range symlinks {
+		_ = os.Symlink(
+			filepath.Join(mountPath, "usr", symlink),
+			filepath.Join(mountPath, symlink))
 	}
+
+	// TODO: use the architecture, not hardcode amd64
+	_, _ = exec.Command("cp", "bin/amd64/id-stub", filepath.Join(mountPath, distro.IdCmd())).CombinedOutput()
+	// TODO: needed for user_group_lookup.c
```

Comment:
The glibc functions used in `user_group_lookup.c` end up invoking `dlopen` for `libnss`. I think this happens [here](https://github.com/bminor/glibc/blob/87faac55161d7dd62997fb09e9fcb5890cf6e4c6/nss/nsswitch.c#L369). Not much we can do about that, so I don't know if the `TODO` is needed here.